pkgs/by-name/de/decasify/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+
+  # nativeBuildInputs
+  zstd,
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+
+  # buildInputs
+  typos,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "decasify";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/alerque/decasify/releases/download/v${finalAttrs.version}/decasify-${finalAttrs.version}.tar.zst";
+    sha256 = "sha256-HTUAb/yL3H4B/n/Ecd/fDpnTYiqwco/E07sa6pFIIU4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    nativeBuildInputs = [ zstd ];
+    dontConfigure = true;
```

Comment:
I can't say this is something I see usually...is there any specific reason for it? If so, a comment would be nice

Reply:
A comment might be appropriate, but perhaps @doronbehar could better explain why this is the way it is. This is a solution we worked out for the `git-warp-time` derivation and also applied to `sile`. I just know it doesn't work right without it. It seems to be a bug in the Nix build system actually because ZIP or GZIP compressed sources are handled differently. The extraction from `fetchurl` happens more or less automatically with the right decompression either way, but `rustPlatform.fetchCargoTarball` doesn't catch a clue and follow suite. If you happen to have downloaded and cached the upstream sources already (e.g. fetched locally to figure out the hash) then a local build will work fine without this bit, but it will fail remotely in CI where the matching hash artifact doesn't exist *before* the build.

Reply:
> but perhaps @doronbehar could better explain why this is the way it is

Yes this is slightly tricky. It seems that by default the derivation that the derivation `fetchCargoTarball` builds (defined [here](https://github.com/NixOS/nixpkgs/blob/e405f30513169feedb64b5c25e7b00242010af58/pkgs/build-support/rust/fetch-cargo-tarball/default.nix#L72-L157)), tries to run the default `./configure` phase that the Nix `stdenv` tries to run. Usually for most Rust projects, there is no `./configure` script and no body has to specify that, but here and both in `sile` and `git-warp-time`, the `src` contains a `./configure` script that shouldn't be run if we just want to prefetch the rust crates...

I wouldn't consider this a bug, but perhaps the default behavior should be changed. And I agree that it would be nice to explain this in a comment, even a short one:

```
# so the cargo fetcher won't try to run the `./configure` script
```

Reply:
Aah, that's making more sense. In that case yes I would definitely suggest that a command `fetchCargTarball` shouldn't be running a configure phase even if it exists. I understand my use of GNU Autotools with Rust projects is a little unusual, but unusual is not wrong and on the flip side I can't think of a single case where a configure script of any kind should be taken into account when figuring the contents of a Cargo sources vendor tarball. In fact in pure Cargo based projects this is never the case either, `cargo fetch` specifically does *not* take into account the current architecture or target triple or any features (all, default, no-std, etc.), it is expected to fetch all possible permutations of dependencies and sort our what gets *used* later in the `cargo build` or `cargo test` or whatever stages when feature flags, platform targets, etc. *can be * specified.

Reply:
I agree: https://github.com/NixOS/nixpkgs/pull/360540